### PR TITLE
Update 'Working with HTTPS' description

### PR DESCRIPTION
**Pull request by s1n4**

### CODE REVIEW

**Comment by pablohoffman** [comment]
couple of things for @rdowinton to process and merge this:

- is it in production already?. do we want to flag it as experimental for the time being? (as we did for sessions)
- maybe we should keep supporting the old way (as stable) while the new one is experimental?

**Comment by qrilka** [comment]
@pablohoffman I think docs should be updated on Crawlera release containing the feature and I agree that we need to stress that it's an experimental feature and the old one should be preferred for production use

File: crawlera.rst [hunk 1]
```diff
@@ -45,15 +45,15 @@ To enable Crawlera in `Dash <http://dash.scrapinghub.com/>`_, see :ref:`crawlera
 Working with HTTPS
 ------------------
 
-The caveat of Crawlera’s HTTP proxy interface is that it doesn’t support the *CONNECT* method, so HTTPS URLs won’t work with a standard proxy configuration. While some HTTP clients support HTTPS requests through standard HTTP proxies, it's recommended you use the :ref:`fetch-api`; however, if you're using the Crawlera middleware it's recommended you use the :ref:`x-crawlera-use-https` header instead.
-
+If you're using the Crawlera middleware it's recommended you use the :ref:`x-crawlera-use-https` header instead. Otherwise, you will need to install
+`Crawlera CA <http://doc.scrapinghub.com/crawlera-ca.crt>` in order to request HTTPS URLs.
```
Comment (dangra) [inline crawlera.rst]: I think the description can be expanded to the 4 possible ways to download HTTPS urls in proxy mode:

1. `x-crawlera-use-https` header
2. `GET https://url.com`  <-- currently used by Scrapy
4. `CONNECT` disabling server certificate verification client side but not possible in browsers. i.e. [requests](http://docs.python-requests.org/en/latest/user/advanced/#ssl-cert-verification)
5. By installing the almighty CA bundle to the system or pointing it from client libraries.

Comment (rdowinton) [inline crawlera.rst]: The certificate method should be mentioned first if it's going to be preferred over X-Crawlera-Use-HTTPS.

**Comment by rdowinton** [comment]
+1 to @qrilka 's suggestion of updating doc upon release and marking as experimental.

**Comment by rdowinton** [comment]
@s1n4 based on Kirill and Daniel's suggestions I would go with something similar to the following:

> Crawlera's use of the CONNECT method is experimental. If you're using the middleware, it's recommended you use the X-Crawlera-HTTPS header, otherwise you should use the fetch API instead.

> If you wish to use the CONNECT method, you will need to install our `certificate <http://doc.scrapinghub.com/crawlera-ca.crt>` or `disable certificate verification <http://docs.python-requests.org/en/latest/user/advanced/#ssl-cert-verification>` in your requests.

**Comment by s1n4** [comment]
Thank you!
Updated.

**Comment by qrilka** [comment]
@rdowinton but why do we say anything about middleware? Did you mean Scrapy middleware? Crawlera is client library agnostic so words about Scrapy should appear in a section with Scrapy examples for Crawlera and not for a generic description of the service.
So I think that we need to remove that "middleware" part

**Comment by rdowinton** [comment]
@qrilka I can't remember the reasoning, but I know in the original doc we recommended use of X-Crawlera-Use-HTTPS when using middleware, and when not using the middleware to use fetch API. I'm guessing the middleware replaces https:// with http:// or something similar, which may be required for use with X-Crawlera-Use-HTTPS.

**Comment by pablohoffman** [comment]
Agree with @qrilka, Crawlera doc should be kept as Scrapy agnostic as possible (only referring to Scrapy as a minor hint somewhere). I liked how @dangra described it in 4 points, can we elaborate on that?

**Comment by s1n4** [comment]
Maybe something like this?
> Crawlera provides three ways for working with HTTPS:
> 1. X-Crawlera-Use-HTTPS header
> 2. Fetch API
> 3. CONNECT method
>
> Crawlera's support of the CONNECT method is experimental. It's recommended you use the X-Crawlera-Use-HTTPS header, otherwise you should use the Fetch API instead.
>
> If you wish to use the CONNECT method, you will need to install our certificate <http://doc.scrapinghub.com/crawlera-ca.crt> or disable certificate verification in your HTTP client.

Any thoughts?

**Comment by rdowinton** [comment]
@s1n4 that looks good, I think it works better with a numbered list. Like I mentioned in my previous comment, if fetch API is still preferred over X-Crawlera-Use-HTTPS for people not using the middleware, then we should make note of that. 

If there's no difference whether using the middleware or not, I would still change the sentence to something like "It's recommended you use either the X-Crawlera-Use-HTTPS header or the fetch API." The 'otherwise' in the existing text implies that fetch API should be a third choice, to be used only if X-Crawlera-Use-HTTPS isn't suitable for you, but it doesn't state why it may be unsuitable (if it is at all).

**Comment by s1n4** [comment]
Updated.

**Comment by rdowinton** [comment]
Cool, I'll merge when changes are deployed.

**Comment by dangra** [comment]
Why not also mentioning the simple proxy protocol that pass full url in
first line also works, I refer to `GET https://google.com` for example.

2015-04-29 12:57 GMT-03:00 Richard Dowinton <notifications@github.com>:

**Comment by qrilka** [comment]
+1 for Daniel's point

**Comment by s1n4** [comment]
> Why not also mentioning the simple proxy protocol that pass full url in
> first line also works, I refer to `GET https://google.com` for example.

In my opinion, that doesn't need to be documented because that's HTTP not a feature or something.

**Comment by dangra** [comment]
In practice nobody expects standard HTTP clients to request https urls
without CONNECT in between, even standard proxies return errors when
receiving such urls. That's why I think it is worth mentioning.

2015-04-30 7:19 GMT-03:00 Sina Samavati <notifications@github.com>:

**Pull request merged**